src/Language/PureScript/Sugar/LetPattern.hs

```diff
@@ -0,0 +1,48 @@
+-- |
+-- This module implements the desugaring pass which replaces patterns in let-in
+-- expressions with appropriate case expressions.
+--
+module Language.PureScript.Sugar.LetPattern (desugarLetPatternModule) where
+
+import Prelude.Compat
+
+import Control.Monad.Error.Class (MonadError(..))
+import Control.Monad.Supply.Class
+
+import Language.PureScript.AST
+import Language.PureScript.Errors
+
+-- |
+-- Replace every @BoundValueDeclaration@ in @Let@ expressions with @Case@
+-- expressions.
+--
+desugarLetPatternModule :: forall m. (MonadSupply m, MonadError MultipleErrors m) => Module -> m Module
```

Comment:
No, you still don't need `MonadError`. `rethrowWithPosition` only applies when you might throw something, which I don't think you do.